File: Problems/12924/Problem.cpp
#include <iostream>
using namespace std;

int solution(int n) {
    int answer = 0;
    
    int sum = 0;
    for (int i=1; i<=n; i++)
    {
        for (int j=i; j<=n; j++)
        {
            //std::cout << j;
            sum += j;
            if (sum == n)
            {
                answer++;
                break;
            }
            else if (sum > n)
            {
                break;
            }
        }
        sum = 0;
    }
    return answer;
}

int main() 
{
    cout << solution(15) << endl;
    
    return 0;
}

File: Problems/42886/Problem.py
from itertools import combinations

def solution(weight):
    #print(sorted(weight)[0:7])
    #for item in list(combinations(sorted(weight)[0:7], 6)):
        #print(sum(item))
    answer = 1
    for w in sorted(weight):
        if answer < w:
            break
        answer += w
    return answer

print(solution([3, 1, 6, 2, 7, 30, 1]))